### ISSUE TITLE
Consider adding a general purpose Link block or block supports

### ISSUE THREAD

**Issue by noisysocks** [WordPress/gutenberg#62262]
To my knowledge it's not currently possible to build a section like this (from [this site](https://aglobewelltravelled.com)) in a block theme:

<img width="1265" alt="Screenshot 2024-06-04 at 15 11 59" src="https://github.com/WordPress/gutenberg/assets/612155/96df1205-adf2-443d-9351-f6312594ec6e">

You can get _tantalisingly_ close using the following structure and setting the Cover block to _Use featured image_ and giving it a gradient overlay.

```
Query Loop
  Post Template
    Cover
      Title
      Excerpt
```

The problem is that you can't make the Cover block link to the post.

It would be great if there were a way to achieve this. I'm not sure of the best solution, but two ideas:

1. A general purpose _Link_ block which wraps its contents with an `<a>` and so allows you to turn anything into a link. This could even replace the existing _Link to post_ and _Link to home_ settings that exists in the Post Featured Image, Post Title, and Site Title blocks. We would need a way to restrict the user from nesting `<a>` tags though as this is invalid markup.

2. Add a new _Link_ block supports that shows up as a panel in the block inspector and add it to lots of different blocks: Cover, Group, Image, etc.

**Comment by andrewserong** [comment]
Nice issue! It reminds me of an earlier issue about making the entire post item within a query loop clickable. Potentially related issues:

* https://github.com/WordPress/gutenberg/issues/38014
* https://github.com/WordPress/gutenberg/issues/60499

**Comment by noisysocks** [comment]
I bet @richtabor has some good ideas here.

**Comment by colorful-tones** [comment]
I would be careful with the final markup. Typically a clickable card solution relies on CSS positioning and a pseudo-element. A good reference for implementation would be Inclusive Components > [Cards](http://inclusive-components.design/cards/)

**Comment by cbirdsong** [comment]
There is a longstanding issue for this over here: 
- #12684

I concur about Inclusive Components being a good reference and that you'd have to be very careful with the markup as well as the editing experience – getting the implementation of this pattern right is _difficult_. 

For instance, in the OP's example screenshot, the whole card is presumably clickable, but there's also a "add comment" link inside of it. The usual recommendation for this pattern is to avoid nested links, but if they have to exist they should have a large clickable interactive area. There is not really a way for that to be applied in the block editor in a generalized way. 

There's also the question of what bit of text actually is linked, which is important for accessibility. [As described in Inclusive Components](https://inclusive-components.design/cards/#:~:text=Now%2C%20all%20of%20the%20card%20contents%20form%20the%20label%20of%20the%20link.%20So%20when%20a%20screen%20reader%20encounters%20it%2C%20the%20announcement%20might%20be%20something%20link%20%22Card%20design%20woes%2C%20ten%20common%20pitfalls%20to%20avoid%20when%20designing%20card%20components%2C%20by%20Heydon%20Pickering%2C%20link%22.), you can't just wrap the whole thing in an `<a>` without creating accessibility issues. Additionally, a common thing you see in this pattern is having a "read more" tease at the bottom of the card, but if that's the linked text it's an instant accessibility failure. The headline needs to be what's linked so it would be unique.

Needless to say, this all would be _very_ difficult for the average user to understand and manage in the editor UI. My inclination would be to make it so that you could only engage "stretched link" option if the container has one link and no other interactive elements inside of it, and once it is a stretched link then additional interactive elements could not be added.

**Comment by mtias** [comment]
Also any solution that changes the root element to `a` is going to be a bit troublesome for block validation and could create too much complexity at the block functions level. It seems a wrapper block could keep things more organized and we could explore enforcing certain restrictions or warnings for nested links or interactive elements. With the support `group` has for alternative tags, maybe it could even be explored as a variation of group.

**Comment by noisysocks** [comment]
Good point on block validation. I could see a "Link to" setting in Group and Cover working pretty well. The complex part is getting the UX right around forbidding nested interactive elements.

**Comment by Marc-pi** [comment]
yep, a strong need indeed see https://github.com/WordPress/gutenberg/issues/12684

**Comment by sophiegyo** [comment]
This would be wonderful if possible! Since I discovered around a year or 18 months ago that the read more block could "hack" this I've used that for example/test work where I wanted the cover block to be clickable - but having the cover itself be a link would be the ideal.

For reference, my terrible, not-accessibility-friendly hack:

* Query loop
  * Post template 
    * Cover block, set to featured image. Inner blocks should **not** use content width. Remove any default padding from the cover block.
      * Read More block, text colour set to fully transparent, and padding jacked up as high as it will go. Optional: text size set to 0px, and customise the text in some manner.
      * Group/stack block with a dark background, slightly opaque
        * Post title
        * Post date

It doesn't let the group containing the Post Title and Post Date be clickable, unfortunately the padding hack doesn't work on those blocks in the same way as the Read More block so only their text is clickable.

[Screenshot link](https://github.com/user-attachments/assets/0724dab9-5a27-4d22-b739-95d053de0690) - showing some earlier setup of the read more block (large text instead of small, but IMO the small text seems to work just as well if not better)

Edit: considering that adding an a block around the cover might be a problem re block validation, would it be viable to instead add it inside, as long as inner blocks don't use content width?
